chemprop/nn/loss.py

```diff
@@ -348,3 +349,30 @@ def _calc_unreduced_loss(self, preds: Tensor, targets: Tensor, mask: Tensor, *ar
 
     def extra_repr(self) -> str:
         return f"threshold={self.threshold}"
+
+
+@LossFunctionRegistry.register(["quantile", "pinball"])
+class QuantileLoss(LossFunction):
+    def __init__(self, task_weights: ArrayLike = 1.0, alpha: float = 0.1):
+        super().__init__(task_weights)
+
+        self.alpha = alpha
+        self.tau_lower = alpha / 2
+        self.tau_upper = 1 - alpha / 2
+
+    def _calc_unreduced_loss(self, preds: Tensor, targets: Tensor, mask: Tensor, *args) -> Tensor:
+        mean, interval = torch.chunk(preds, 2, 1)
```

Comment:
broader question here: because we don't currently support uncertainty prediction, we can change the current object model. Currently, we output a tensor of shape $(b, t*s)$, where $b$ is the batch size, $t$, the number of tasks, and $s$ the number of uncertainty targets. It makes it a little awkward to unpack predictions, as is the case here, where we have to chunk the raw tensor. What do we think about changing the output shape of uncertainty-based model to $(b, t, s)$, so that we have a separate dimension of the tensor for each concept: input, task, and uncertainty parameter?

Reply:
The number of outputs for the FFN is $t * s$. Thus, I think we have to chunk the tensor in forward() of the predictor. I guess you mean to replace the output of `torch.cat((mean, v, alpha, beta), 1)` with `torch.stack((mean, v, alpha, beta), dim=2)` in the case of `EvidentialFFN`? Anyhow, I think it would be a good idea to separate those prediction results into separate dimensions. It can help us more easily understand the output of MveFFN, EvidentialFFN, BinaryDirichletFFN, and QuantileFFN.

Reply:
Exactly, placing separate concepts into separate dimensions, ex: $\hat y_{ijk}$, where $i$ indexes the input, $j$, the input, and $k$ the uncertainty parameter. So if someone wants only the mean predictions, I think it's clearer to write `preds[:, :, 0]` rather than `preds[:, ::s]`, where `s` is the number of uncertainty parameters and needs to be known somehow. It's not a performance difference, but I just think it's clearer.

Reply:
I love this idea! It makes handling predictions a lot easierm and also make the code easier to understand. I will work on https://github.com/chemprop/chemprop/pull/959 accordingly.